### PR TITLE
Django filter uplift

### PR DESCRIPTION
**Pull request by Pomax**
builds on top of https://github.com/mozilla/network-pulse-api/pull/545 (review after reviewing that one) as django-filters 2.x relies on DRF 3.7+

### CODE REVIEW

**Comment by gideonthomas** [comment]
I'll need a little bit more time with this. I remember there being some weirdness around the way we did the Profile filter that would be affected by an upgrade in the filter class, but I don't remember exactly what that was so I'll need to investigate a bit.

**Comment by Pomax** [comment]
temporarily assigning to @gideonthomas for that investigation

**Comment by Pomax** [comment]
https://github.com/mozilla/network-pulse-api/pull/545 landed, this PR can now be rebased and the PR comments worked in.

File: pulseapi/profiles/views/profiles.py [hunk 1]
```diff
@@ -104,11 +106,7 @@ class NumberInFilter(django_filters.BaseInFilter, django_filters.NumberFilter):
     pass
 
 
-# NOTE: DRF has deprecated the FilterSet class in favor of
-# django_filters.rest_framework.FilterSet in v3.7.x, which
-# we aren't far from upgrading to.
-# SEE: https://github.com/mozilla/network-pulse-api/issues/288
-class ProfileCustomFilter(filters.FilterSet):
+class ProfileCustomFilter(FilterSet):
```
[the remark this answers is not in the document]
Comment (Pomax) [reply pulseapi/profiles/views/profiles.py]: 1. :+1:
2. hmm... the `list(chain(....))` being returned is something that's usually used for queries with incompatible database results, but in this case it looks like we _should_ just be able to say  `return qs_startswith | qs_contains`, using the set union (not bitwise or) operator.

I'll see if that works and update the PR accordingly

**Pull request merged**